liblustreapi/src/lib.rs

```diff
@@ -265,29 +313,84 @@ impl Llapi {
         Ok(())
     }
 
+    // Does llapi_rmfid() or returns back fidlist as a Vec<String> in Err()
+    fn llapi_rmfid(
+        &self,
+        mntpt: &str,
+        fidlist: impl IntoIterator<Item = String>,
+    ) -> Result<(), Vec<String>> {
+        let flist: Vec<String> = fidlist.into_iter().collect();
+        let func: lib::Symbol<extern "C" fn(*const libc::c_char, *const RmfidsArg) -> libc::c_int> =
+            match unsafe { self.lib.get(b"llapi_rmfid\0") } {
+                Err(e) => {
+                    log::info!("Failed load llapi_rmfid: {:?}", e);
+                    return Err(flist);
+                }
+                Ok(f) => f,
+            };
+
+        let devptr = match CString::new(mntpt) {
+            Ok(c) => c.into_raw(),
+            Err(_) => return Err(flist),
+        };
+
+        let mut fids: Vec<RmfidsArg> = Vec::with_capacity(flist.len() + 1);
+        fids.push(FidArrayHdr::new(0).into());
+        let flist2 = flist.clone();
+        for fidstr in flist {
```

Comment:
One thing to keep in mind is that for loops can also be written as fold's: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.fold